### PR TITLE
feat: configure fips for rhel 8

### PR DESCRIPTION
**Pull request by faiq**
**What problem does this PR solve?**:

**Which issue(s) does this PR fix?**:

* https://d2iq.atlassian.net/browse/D2IQ-93017

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: ansible/roles/fips/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+- name: "RHEL 7 FIPS setup"
```
Comment (dkoshkin) [inline ansible/roles/fips/tasks/main.yaml:1]: Please add a link here too

File: ansible/roles/fips/tasks/redhat-7.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    path: /etc/sysconfig/prelink
+    regexp: '^PRELINKING=yes'
+    replace: '^PRELINKING=no'
+  when:
+    - sysfips is changed
+
+- name: "redhat 7 fips: install dracut fips"
+  yum:
+    name:
+      - dracut-fips
```
Comment (dkoshkin) [inline ansible/roles/fips/tasks/redhat-7.yaml:21]: This package is not in the bundle, maybe we should only do this for >RHEL 8?

File: ansible/roles/fips/tasks/redhat-8.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+- name: "redhat 8 fips: run fips enable command"
+  command: fips-mode-setup --enable
+  when:
+    - sysfips is changed
+
+- name: "redhat 8 fips: reboot to pick up fips changes"
+  ansible.builtin.reboot:
+  when:
+    - sysfips is changed
+    - fips_reboot
```
Comment (dkoshkin) [inline ansible/roles/fips/tasks/redhat-8.yaml:19]: I don't think it makes sense to change the FIPS mode but not reboot. IMO it should be all or nothing.

File: overrides/fips.yaml [hunk 4]
```diff
@@ -4,6 +4,8 @@ k8s_image_registry: docker.io/mesosphere
 fips:
   enabled: true
```
Comment (dkoshkin) [inline overrides/fips.yaml:5]: Or something like that
```suggestion
fips:
  enabled: true
  # when set to true and FIPS mode is not already enabled, FIPS mode will be enabled and the machine will be rebooted
  # only works on RHEL 8 
  configure: false
```

File: ansible/roles/fips/tasks/redhat-7.yaml (hunk 2 above)
Comment (faiq) [reply ansible/roles/fips/tasks/redhat-7.yaml:21]: we're doing it for rhel 7 thats it. i still haven't worked on testing this bit

File: overrides/fips.yaml (hunk 4 above)
Comment (faiq) [reply overrides/fips.yaml:5]: yes actually i love this. 

we should guard all the tasks in `fips` role  to only run if `configure == true`

File: ansible/roles/fips/tasks/redhat-7.yaml (hunk 2 above)
Comment (dkoshkin) [reply ansible/roles/fips/tasks/redhat-7.yaml:21]: I meant not have this codepath at all for RHEL 7. Only do https://github.com/mesosphere/konvoy-image-builder/pull/508/files#diff-5f48c283af385dc86e531a1d0821000e9fee1c77cc9755ee4cc4c3f2c5b0eb76R1 for RHEL 8.

File: overrides/fips.yaml (hunk 4 above)
Comment (dkoshkin) [reply overrides/fips.yaml:5]: Looks like you forgot to make this change too.
You will also need to override this configuration in KIB's CI.

Comment (faiq) [reply overrides/fips.yaml:5]: right now im keeping this bit locally and for CI I'm planning on doing it in an extra override passed  to here.

File: ansible/roles/fips/tasks/redhat-7.yaml (hunk 2 above)
Comment (faiq) [reply ansible/roles/fips/tasks/redhat-7.yaml]: removed

File: ansible/roles/fips/tasks/redhat-8.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+- name: "redhat 8 fips: run fips enable command"
+  command: fips-mode-setup --enable
+  when:
+    - sysfips is changed
+
+- name: "redhat 8 fips: reboot to pick up fips changes"
+  ansible.builtin.reboot:
+  when:
+    - sysfips is changed
```
Comment (supershal) [inline ansible/roles/fips/tasks/redhat-8.yaml:18]: nit: `sysfips is changed` status check is valid. wdyt adding another condition to check success status of `fips-mode-setup --enable` task

Comment (supershal) [reply ansible/roles/fips/tasks/redhat-8.yaml:18]: please disregard above. the reboot task will proceed only if `fips-mode-setup --enable` is successful. no need to unnecessary check for its success.

**Review by supershal — APPROVED** [review]

File: overrides/fips.yaml (hunk 4 above)
Comment (faiq) [reply overrides/fips.yaml:5]: added these to CI. and made the default false

**Review by dkoshkin — APPROVED** [review]

File: ansible/roles/fips/tasks/main.yaml [hunk 6]
```diff
@@ -0,0 +1,8 @@
+# https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-the-system-wide-cryptographic-policies_security-hardening#switching-the-system-to-fips-mode_using-the-system-wide-cryptographic-policies
```
Comment (faiq) [inline ansible/roles/fips/tasks/main.yaml:1]:
```suggestion
# see: https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/security_hardening/using-the-system-wide-cryptographic-policies_security-hardening#switching-the-system-to-fips-mode_using-the-system-wide-cryptographic-policies
```

**Pull request merged**